lib/sort_by_length.rb

```diff
@@ -1,7 +1,27 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(nlog(n)) where n is the length of the string
+# Space complexity: o(n) where n is the length of the string
 def sort_by_length(my_sentence)
```

Comment:
👍 This works and has really nice recursion, but you do have a bit of an issue with time complexity (see below).  It's better if you use index numbers to keep track of your current location in the left and right subarrays.